Add Terraform config for patches

igorveselinovic opened the pull request:
Resolves #13 

The most significant new thing here is TimescaleDB. This uses a new module defined in `bespin` that creates an ECS cluster with a single EC2 instance and a single container running TimescaleDB on that instance.

The full list of dependencies is:
- Kafka (AWS MSK)
- TimescaleDB (our own module)
- `heimdall`
- `ether`
- `karen` (not a direct dependency for `patches`, but it's needed for `ether`)
- RDS (needed for `ether` and `karen`)
- EFS (needed for `ether`)

So yeah, basically the whole (current) backend is being deployed here. I've manually tested the frontend against `patches` (protected by `heimdall`) after setting up a user with `karen` and a conversation with `ether` and that whole flow is working.

igorveselinovic committed:
Update Makefile with docker registry stuff

igorveselinovic committed:
Add initial terraform code

Patches and TimescaleDB modules plus integration environment

igorveselinovic committed:
Add cert creation to Makefile

Since we deploy Heimdall in the integration environment

igorveselinovic committed:
Add Kafka config and hook patches up to heimdall

Patches is accessible directly or indirectly through heimdall

igorveselinovic committed:
Update patches tf config

* use less memory
* fix environment variable names
* make Classic Load Balancer forward TCP traffic so that WebSockets can
  work

igorveselinovic committed:
Fix module sources to use github repos

igorveselinovic committed:
Update timescaledb module to refer to GitHub

igorveselinovic committed:
Update Kafka brokers to run in private subnets

Thao-Tran commented on terraform/main.tf:
```diff
@@ -0,0 +1,264 @@
+  vpc_id      = module.ecs_base.vpc_id
+
+  ingress {
+    from_port       = 2049
```
What do you mean you've never seen Blade Runner?

igorveselinovic replied:
❄️🐵

igorveselinovic committed:
Update tf to fix Heimdall internal server access

It needs its own cluster to so that it is running on a separate EC2
instance from everything else. It needs to do this because of the way
Network Load Balancers work: they don't like requests looping back to
the same instance.

Also updated the Heimdall module config to use the new variables.
